src/System.Private.CoreLib/src/System/Threading/ThreadPool.cs

```diff
@@ -566,6 +570,17 @@ internal static bool Dispatch()
                     //
                     workQueue.EnsureThreadRequested();
 
+                    // Start on clean ExecutionContext and SynchronizationContext
```

Comment:
Would it be better to clear the contexts after a work item such that they can be collected if there are no more work items and the thread pool goes idle?

Reply:
I originally had that but the Assert failed that the EC wasn't clean; at best I can tell unmanged items also use the threadpool threads and one of the timers from the `AppDomainTimerCallback` doesn't run from the `Dispatch` loop, but uses the thread its on (that does the dispatching to the ThreadPool) to run the last timer; so one of those might have "leaked" a change.

Reply:
Makes sense, but the values can be cleared once before the dispatch loop and thereafter after each work item completes. Don't think it would make a big difference but maybe slightly better.

Reply:
Ah, good idea